libnmstate/netapplier.py

```diff
@@ -315,7 +318,35 @@ def assert_ifaces_state(ifaces_desired_state, ifaces_current_state):
 
         if iface_dstate != iface_cstate:
             raise DesiredStateIsNotCurrentError(
-                'desired: {}\ncurrent: {}'.format(iface_dstate, iface_cstate))
+                _format_desired_current_state_diff(iface_dstate,
+                                                   iface_cstate)
+            )
+
+
+def _format_desired_current_state_diff(desired_state, current_state):
```

Comment:
How about moving this to prettystate? Calling it using prettystate.diff_state(...) fits nicely.

Reply:
This will require to make the labels "desired", "current" for the state part of the parameters. How about we change this when there is another user for the differ that makes it clearer what parameters we need.

Reply:
changed this